src/core/reactor_backend.cc

```diff
@@ -1146,7 +1146,13 @@ bool reactor_backend_selector::has_enough_aio_nr() {
      * So this method calculates:
      *  Available AIO on the system - (request AIO per-cpu * ncpus)
      */
-    if (aio_max_nr - aio_nr < reactor::max_aio * smp::count) {
+    unsigned _smp_local_count = 1;
+
+    if (smp::count > 0){
+       _smp_local_count = smp::count;
+    }
+
```

Comment:
I don't understand how the above is going to resolve the problem at hand.
I believe that the only acceptable value you need is a real (!!) smp::count, e.g. 64 on a machine with 64 CPUs. Having 1 instead as your code would do is not addressing the issue AFAIU.

Reply:
The proper fix IMO is to make the original code run when smp::count evaluates to the expected value.
Using smp::count before it's initialized is simply a bug.

Reply:
This is somehow documented a couple lines above -- I think what was unexpected is that we initialize `smp::count = 0`. That's the relevant comment already within the code:

```cpp
    /* reactor_backend_selector::available() will be execute in early stage,
     * it's before io_setup() issued, and not per-cpu basis.
     * So this method calculates:
     *  Available AIO on the system - (request AIO per-cpu * ncpus)
     */
```

TLDR; the backend selection happens early before we eventually parse the user command-line at `smp::configure`, so that means we initialize `smp::count = 0` and this is what causes the validation error. 

So that gives us two options:

1. Stick with the proposed solution here or,
2. Initialize `smp::count = 1` as it makes no sense - indeed - to start it with 0 (we can't run seastar in a pencil yet AFAIU)

Let me know which option seems better and I will push it with the relevant changes.

Reply:
@xemul Any comments?